### PR TITLE
Upload project from GitHub

### PR DESCRIPTION
**Pull request by babsey**
This PR allows user to import projects from github repo (nest-desktop/nest-desktop-projects).
Plus the term (download/upload) of projects is renamed to export/import.

The doc is updated in other branch (https://github.com/babsey/nest-desktop/tree/projects-import-export-doc).

### CODE REVIEW

File: src/components/project/ProjectsImportDialog.vue [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+      items: [
+        {
+          icon: 'mdi-paperclip',
+          title: 'Load projects from file',
+          value: 'file',
+        },
+        {
+          icon: 'mdi-github',
+          title:
+            'Load projects from github repo (nest-desktop/nest-desktop-projects)',
```
Comment (jens-br) [inline src/components/project/ProjectsImportDialog.vue:174]:
```suggestion
            'Load projects from GitHub repo (nest-desktop/nest-desktop-projects)',
```

File: src/components/project/ProjectsImportDialog.vue [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+          value: 'github',
+        },
+        { icon: 'mdi-web', title: 'Load projects from url', value: 'url' },
```
Comment (jens-br) [inline src/components/project/ProjectsImportDialog.vue:177]:
```suggestion
        { icon: 'mdi-web', title: 'Load projects from URL', value: 'url' },
```

File: src/components/project/ProjectsImportDialog.vue [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+  <div class="ProjectsLoadDialog">
+    <v-dialog v-model="state.dialog" max-width="1024">
+      <v-card>
+        <v-card-title v-text="'Import projects'" />
+        <v-card-subtitle v-text="'Select a source'" />
+
+        <v-card-text>
+          <v-row class="mb-1">
+            <v-col cols="2">
+              <v-btn-toggle
```
Comment (jens-br) [inline src/components/project/ProjectsImportDialog.vue:11]: This view seems somehow glitched: The icon for the selected source overlaps with the other button and the selected button is not clearly visible any more. Could you check it again?
grafik

File: src/components/project/ProjectsImportDialog.vue [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+                dark
+                dense
+                mandatory
+                v-model="state.source"
+              >
```
Comment (jens-br) [inline src/components/project/ProjectsImportDialog.vue:16]: I receive the suggestion
```suggestion
              <v-btn-toggle dark dense mandatory v-model="state.source">
```
by ESLint/Prettier. Could you check that?

**Review by jens-br — CHANGES_REQUESTED** [review]
Small adjustments suggested.

File: src/components/project/ProjectsImportDialog.vue (hunk 3 above)
Comment (babsey) [reply src/components/project/ProjectsImportDialog.vue:11]: It is strange, in my case it looks clean.

image

Comment (jens-br) [reply src/components/project/ProjectsImportDialog.vue]: Indeed. I retried it with a clean Firefox Profile, but there were no changes. Did you introduce some changes on your machine, e.g. updates of packages, etc.?

File: src/components/project/ProjectsImportDialog.vue [hunk 5]
```diff
@@ -0,0 +1,317 @@
...
+                  :key="item.value"
+                  :title="item.title"
+                  :value="item.value"
+                  v-for="item in state.items"
+                >
+                  <v-icon v-text="item.icon" />
+                </v-btn>
+              </v-btn-toggle>
+            </v-col>
+            <v-col class="pa-3" cols="10">
```
Comment (jens-br) [inline src/components/project/ProjectsImportDialog.vue:22]:
```suggestion
            <v-col cols="auto">
              <v-btn-toggle dark dense mandatory v-model="state.source">
                <v-btn
                  :key="item.value"
                  :title="item.title"
                  :value="item.value"
                  v-for="item in state.items"
                >
                  <v-icon v-text="item.icon" />
                </v-btn>
              </v-btn-toggle>
            </v-col>
            <v-col class="pa-3" cols="9">
```
grafik
or at least
```suggestion
            <v-col cols="3">
              <v-btn-toggle dark dense mandatory v-model="state.source">
                <v-btn
                  :key="item.value"
                  :title="item.title"
                  :value="item.value"
                  v-for="item in state.items"
                >
                  <v-icon v-text="item.icon" />
                </v-btn>
              </v-btn-toggle>
            </v-col>
            <v-col class="pa-3" cols="9">
```

grafik


seemed to work better. What does this look like when you use your setup?

File: src/components/project/ProjectsImportDialog.vue (hunk 3 above)
Comment (babsey) [reply src/components/project/ProjectsImportDialog.vue]: I cleaned site data and upgraded packages, but my button group is still smaller  than yours.

File: src/components/project/ProjectsImportDialog.vue [hunk 6]
```diff
@@ -0,0 +1,317 @@
+<template>
+  <div class="ProjectsLoadDialog">
+    <v-dialog v-model="state.dialog" max-width="1024">
+      <v-card>
+        <v-card-title v-text="'Import projects'" />
+        <v-card-subtitle v-text="'Select a source'" />
+
+        <v-card-text>
+          <v-row class="mb-1">
+            <v-col cols="3">
```
Comment (jens-br) [inline src/components/project/ProjectsImportDialog.vue:10]:
```suggestion
            <v-col :cols="state.items.length">
```

File: src/components/project/ProjectsImportDialog.vue [hunk 7]
```diff
@@ -0,0 +1,317 @@
...
+                  :key="item.value"
+                  :title="item.title"
+                  :value="item.value"
+                  v-for="item in state.items"
+                >
+                  <v-icon v-text="item.icon" />
+                </v-btn>
+              </v-btn-toggle>
+            </v-col>
+            <v-col class="pa-3" cols="9">
```
Comment (jens-br) [inline src/components/project/ProjectsImportDialog.vue:22]:
```suggestion
            <v-col class="pa-3" :cols="12 - state.items.length">
```

**Review by jens-br — COMMENTED** [review]
That works smoothly, I'd only like to keep the relation between state.items.length and the column number.

**Comment by jens-br** [comment]
As discussed, let's keep it for now.

**Review by jens-br — APPROVED** [review]

**Pull request merged**